osf/migrations/0038_abstractnode_root.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9 on 2017-02-28 20:20
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
```

Comment:
Good idea. Alternatively, I could add a check in the post save handler to ensure that root is always set:

```python
if created or not instance.root:
     instance.get_root()
```